debezium-core/src/main/java/io/debezium/pipeline/signal/actions/AbstractSnapshotSignal.java

```diff
@@ -20,9 +24,20 @@ public abstract class AbstractSnapshotSignal<P extends Partition> implements Sig
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractSnapshotSignal.class);
     protected static final String FIELD_DATA_COLLECTIONS = "data-collections";
+    protected static final String FIELD_DATA_COLLECTION = "data-collection";
     protected static final String FIELD_TYPE = "type";
+    @Deprecated
     protected static final String FIELD_ADDITIONAL_CONDITION = "additional-condition";
+    protected static final String FIELD_ADDITIONAL_CONDITIONS = "additional-conditions";
     protected static final String FIELD_SURROGATE_KEY = "surrogate-key";
+    protected static final String FIELD_FILTER = "filter";
+
+    protected static Set<Pattern> getDataCollectionPatterns(List<String> dataCollections) {
```

Comment:
Doe it mean that the user cannot prescribe order in which tables will be snapshotted?

Reply:
Could you please dig a bit deeper into initial snapshot? It seems to me that the order is driven by the `table.include.list` property.
For the increment snapshot  - yes deduplication of matchin tables should be done - could you please create a se parate Jira for it?

Reply:
> Could you please dig a bit deeper into initial snapshot? It seems to me that the order is driven by the `table.include.list` property. 

```java
private void determineCapturedTables(RelationalSnapshotContext<P, O> ctx, Set<Pattern> dataCollectionsToBeSnapshotted) throws Exception {
        Set<TableId> allTableIds = getAllTableIds(ctx);
        Set<TableId> snapshottedTableIds = determineDataCollectionsToBeSnapshotted(allTableIds, dataCollectionsToBeSnapshotted).collect(Collectors.toSet());

        Set<TableId> capturedTables = new HashSet<>();
        Set<TableId> capturedSchemaTables = new HashSet<>();

        for (TableId tableId : allTableIds) {
            if (connectorConfig.getTableFilters().eligibleForSchemaDataCollectionFilter().isIncluded(tableId)) {
                LOGGER.info("Adding table {} to the list of capture schema tables", tableId);
                capturedSchemaTables.add(tableId);
            }
        }

        for (TableId tableId : snapshottedTableIds) {
            if (connectorConfig.getTableFilters().dataCollectionFilter().isIncluded(tableId)) {
                LOGGER.trace("Adding table {} to the list of captured tables for which the data will be snapshotted", tableId);
                capturedTables.add(tableId);
            }
            else {
                LOGGER.trace("Ignoring table {} for data snapshotting as it's not included in the filter configuration", tableId);
            }
        }

        ctx.capturedTables = addSignalingCollectionAndSort(capturedTables);
        ctx.capturedSchemaTables = capturedSchemaTables
                .stream()
                .sorted()
                .collect(Collectors.toCollection(LinkedHashSet::new));
    }
```

All tables are read from the database and then filtered by the pattern specified in the `snapshot.include.collection.list` properties if provided, otherwise all table will be returned. After that only tables that are includes in `table.include.list` will be used for the snapshot. 

>For the increment snapshot - yes deduplication of matchin tables should be done - could you please create a se parate Jira for it?

Yes.